src/world/ObjectMgr.cpp

```diff
@@ -3898,4 +3898,172 @@ void ObjectMgr::LoadAreaTrigger()
     while (result->NextRow());
 
     Log.Success("AreaTrigger", "Loaded %u area trigger teleport definitions", count);
+}
+
+void ObjectMgr::LoadEventScripts()
+{
+    Log.Notice("ObjectMgr", "Loading Event Scripts...");
+
+    QueryResult* result = WorldDatabase.Query("SELECT * FROM event_scripts WHERE event_id > 0 ORDER BY event_id");
+
+    if (!result)
+    {
+        Log.Notice("ObjectMgr", "Loaded 0 event_scripts. DB table `event_scripts` is empty.");
+        return;
+    }
+
+    uint32 count = 0;
+    do
+    {
+        Field* fields = result->Fetch();
+
+        uint32 event_id = fields[0].GetUInt32();
+        simpleEventScripts eventscript;
+
+        eventscript.eventId = event_id;
+        eventscript.function = ScriptCommands(fields[1].GetUInt32());
+        eventscript.scripttype = EasyScriptTypes(fields[2].GetUInt32());
+        eventscript.data_1 = fields[3].GetUInt32();
+        eventscript.data_2 = fields[4].GetUInt32();
+        eventscript.data_3 = fields[5].GetUInt32();
+        eventscript.data_4 = fields[6].GetUInt32();
+        eventscript.data_5 = fields[7].GetUInt32();
+        eventscript.x = fields[8].GetUInt32();
+        eventscript.y = fields[9].GetUInt32();
+        eventscript.z = fields[10].GetUInt32();
+        eventscript.o = fields[11].GetUInt32();
+        eventscript.delay = fields[12].GetUInt32();
+        eventscript.nextevent = fields[13].GetUInt32();
+
+        simpleEventScripts* sa = &mEventScriptMaps.insert(EventScriptMaps::value_type(event_id, eventscript))->second;
+
+        // for search by spellid ( data_1 is spell id )
+        if (eventscript.data_1 && eventscript.scripttype == SCRIPT_TYPE_SPELL_EFFECT)
+            mSpellEffectMaps.insert(SpellEffectMaps::value_type(eventscript.data_1, sa));
+
+
+        ++count;
+    } while (result->NextRow());
+
+    Log.Success("ObjectMgr", "Loaded event_scripts for %u events...", count);
```

Comment:
This line is redundant with line 3911 (both lines output the number of entries in the table). Try to refactor your code to use a single statement.